### PR TITLE
Jackson media support for Níma

### LINKED ISSUE #6418: Níma: Add Jackson media support
We have JSON-B and JSON-P, but we still miss Jackson for JSON media support for Níma

### PR DESCRIPTION
**Pull request by tomas-langer**
Unit tests for all JSON media supports for Níma

This PR is Níma specific.

Resolves #6418

### CODE REVIEW

File: nima/http/media/jackson/src/test/java/io/helidon/nima/http/media/jackson/JacksonMediaTest.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+import org.junit.jupiter.api.Test;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+/*
+Update the following tests together (if relevant):
```
Comment (romain-grecourt) [inline nima/http/media/jackson/src/test/java/io/helidon/nima/http/media/jackson/JacksonMediaTest.java:44]: Is this a left-over comment ? Why should the mentioned tests be updated together ?

File: common/http/src/main/java/io/helidon/common/http/HttpMediaType.java [hunk 2]
```diff
@@ -191,6 +193,19 @@ default HttpMediaType withCharset(String charset) {
...
+     * Create a new {@link io.helidon.common.http.HttpMediaType} instance with the same type, subtype and parameters
+     * copied from the original instance and the supplied {@value #CHARSET_PARAMETER} parameter.
+     *
+     * @param charset the {@value #CHARSET_PARAMETER} parameter value
+     * @return copy of the current {@code MediaType} instance with the {@value #CHARSET_PARAMETER}
+     *         parameter set to the supplied value.
+     */
+    default HttpMediaType withCharset(Charset charset) {
+        Objects.requireNonNull(charset);
+        return withCharset(charset.name());
```
Comment (ljnelson) [inline common/http/src/main/java/io/helidon/common/http/HttpMediaType.java:206]: `charset.name()` will trigger a `NullPointerException` for you that is "smart"; no need for `Objects.requireNonNull(charset)`

File: nima/http/media/jackson/src/main/java/io/helidon/nima/http/media/jackson/JacksonMediaSupportProvider.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    private JacksonReader reader;
+    private JacksonWriter writer;
+    private ObjectMapper objectMapper;
+
+    @Override
+    public void init(MediaContext context) {
+        objectMapper = new ObjectMapper()
+                .registerModule(new ParameterNamesModule())
+                .registerModule(new Jdk8Module())
+                .registerModule(new JavaTimeModule());
```
Comment (ljnelson) [inline nima/http/media/jackson/src/main/java/io/helidon/nima/http/media/jackson/JacksonMediaSupportProvider.java:52]: Suppose I, as an end developer, wish to customize the `ObjectMapper` in use (a very common thing to do with Jackson). Is my only option to supply a different implementation of `MediaSupportProvider`?

File: nima/http/media/jackson/src/main/java/io/helidon/nima/http/media/jackson/JacksonMediaSupportProvider.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+            }
+        }
+
+        return ReaderResponse.unsupported();
+    }
+
+    @Override
+    public <T> WriterResponse<T> writer(GenericType<T> type, WritableHeaders<?> requestHeaders) {
+        if (requestHeaders.contains(Http.Header.CONTENT_TYPE)) {
+            if (requestHeaders.contains(CONTENT_TYPE_JSON)) {
```
Comment (ljnelson) [inline nima/http/media/jackson/src/main/java/io/helidon/nima/http/media/jackson/JacksonMediaSupportProvider.java:119]: I'm not familiar with all the rules and responsibilities of a `MediaProvider` but do bear in mind Jackson can read/write everything from JSON to properties to YAML. Naïvely this check looks to be too restrictive.

File: nima/http/media/jackson/src/main/java/io/helidon/nima/http/media/jackson/JacksonMediaSupportProvider.java (hunk 3 above)
Comment (tomas-langer) [reply nima/http/media/jackson/src/main/java/io/helidon/nima/http/media/jackson/JacksonMediaSupportProvider.java:52]: Right now, yes. We have an issue to separate the Provider from Support, then we can do a custom instance.

Comment (tomas-langer) [reply nima/http/media/jackson/src/main/java/io/helidon/nima/http/media/jackson/JacksonMediaSupportProvider.java:52]: https://github.com/helidon-io/helidon/issues/6421

File: nima/http/media/jackson/src/test/java/io/helidon/nima/http/media/jackson/JacksonMediaTest.java (hunk 1 above)
Comment (tomas-langer) [reply nima/http/media/jackson/src/test/java/io/helidon/nima/http/media/jackson/JacksonMediaTest.java]: Updated the comment (I hope it is clear now)

File: common/http/src/main/java/io/helidon/common/http/HttpMediaType.java (hunk 2 above)
Comment (tomas-langer) [reply common/http/src/main/java/io/helidon/common/http/HttpMediaType.java:205]: Removed the check.

**Review by romain-grecourt — APPROVED** [review]

**Pull request merged**